conanfile.py

```diff
@@ -1,351 +1,137 @@
-        self.options["qt"].with_md4c = False
-        self.options["qt"].with_mysql = False
-        self.options["qt"].with_odbc = False
-        self.options["qt"].with_openal = False
-        self.options["qt"].with_pq = False
         self.options["qt"].openssl = not is_apple_os(self)
-        if self.settings.os == "iOS" or self.settings.os == "Android":
+        if self._isMobilePlatform:
             self.options["qt"].opengl = "es2"
-        if not is_apple_os(self) and self.settings.os != "Android" and cross_building(self):
-            self.options["qt"].cross_compile = self.env["CONAN_CROSS_COMPILE"]
-        # TODO: add for all platforms after updating recipe
-        if self.settings.os == "Android":
-            self.options["qt"].essential_modules = False
-        # No Qt OpenGL for cross-compiling for Windows, Conan does not support it
-        if self.settings.os == "Windows" and cross_building(self):
-            self.options["qt"].opengl = "no"
 
-        # transitive deps
-        # doesn't link to overridden bzip2 & zlib, the tool isn't needed anyway
-        self.options["pcre2"].build_pcre2grep = False
-        # executable not needed
-        if self.settings.os == "Android":
-            self.options["sqlite3"].build_executable = False
+        # static Qt for iOS is the only viable option at the moment
+        if self.settings.os == "iOS":
+            self.options["qt"].shared = False
 
-    def requirements(self):
-        self.requires("freetype/[~2.12.1]", override=True) # sdl_ttf / Qt
-        self.requires("libpng/[~1.6.39]", override=True) # freetype / qt / sdl_image
+        # MSVC static build requires static runtime, but VCMI uses dynamic runtime
+        if is_msvc(self):
+            self.options["ffmpeg"].shared = True
 
+    def requirements(self):
         # client
         if self.options.with_ffmpeg:
             self.requires("ffmpeg/[>=4.4]")
 
+        # On Android SDL version must be the same as the version of Java wrapper for SDL in VCMI source code
+        # Wrapper can be found in the following directory: android/vcmi-app/src/main/java/org/libsdl/app
+        # TODO: try enabling version range once there's no conflict
+        # sdl_image & sdl_ttf depend on earlier version
+        # ERROR: Version conflict: Conflict between sdl/2.28.5 and sdl/2.28.3 in the graph.
+        # Conflict originates from sdl_mixer/2.8.0
+        # upcoming SDL version 3.0+ is not supported at the moment due to API breakage
+        # SDL versions between 2.22-2.26.1 have broken sound
+        # self.requires("sdl/[^2.26.1 || >=2.0.20 <=2.22.0]")
+        # versions before 2.30.7 don't build for Android with NDK 27: https://github.com/libsdl-org/SDL/issues/9792
+        self.requires("sdl/2.30.9", override=True)
+
         # launcher
         if self.settings.os == "Android":
-            self.requires("qt/[~5.15.14]")
+            self.requires("qt/[~5.15.14]") # earlier versions have serious bugs
```

Comment:
Obviously not urgent, but any plans to upgrade to Qt 6?

Reply:
yes, that's exactly my next milestone, at least for android-64. Other platforms should have it as an option.